From 649841af39b166f4b35f41f916e9c8d384442d7b Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Thu, 16 May 2024 15:54:36 +0200
Subject: [PATCH] Permit to avoid to use hunter for a subset of dependencies

---
 cmake/depthaiDependencies.cmake | 32 ++++++++++++++++++++++----------
 1 file changed, 22 insertions(+), 10 deletions(-)

diff --git a/cmake/depthaiDependencies.cmake b/cmake/depthaiDependencies.cmake
index 9be00b7f6..210a8862e 100644
--- a/cmake/depthaiDependencies.cmake
+++ b/cmake/depthaiDependencies.cmake
@@ -8,22 +8,34 @@ if(CONFIG_MODE)
     set(CMAKE_PREFIX_PATH "${CMAKE_CURRENT_LIST_DIR}/${_IMPORT_PREFIX}" ${CMAKE_PREFIX_PATH})
     set(_QUIET "QUIET")
 else()
+    # Permit to avoid to use hunter for some dependencies
+    # Note: if you need to pass COMPONENTS to hunter_add_package,
+    # this macro needs to be changed
+    macro(depthai_hunter_add_package depthai_dep_package)
+        if(DEFINED HUNTER_SKIP_PACKAGE_${depthai_dep_package} 
+           AND HUNTER_SKIP_PACKAGE_${depthai_dep_package})
+            message(STATUS "Skipping depthai-core dependency ${depthai_dep_package} as HUNTER_SKIP_PACKAGE_${depthai_dep_package} is ENABLED")
+        else()
+            hunter_add_package(${package})
+        endif()
+    endmacro()
+
     set(DEPTHAI_SHARED_LIBS ${BUILD_SHARED_LIBS})
-    hunter_add_package(nlohmann_json)
+    depthai_hunter_add_package(nlohmann_json)
     if(NOT DEPTHAI_XLINK_LOCAL)
-        hunter_add_package(XLink)
+        depthai_hunter_add_package(XLink)
     endif()
-    hunter_add_package(BZip2)
-    hunter_add_package(FP16)
-    hunter_add_package(libarchive-luxonis)
-    hunter_add_package(spdlog)
-    hunter_add_package(ZLIB)
+    depthai_hunter_add_package(BZip2)
+    depthai_hunter_add_package(FP16)
+    depthai_hunter_add_package(libarchive-luxonis)
+    depthai_hunter_add_package(spdlog)
+    depthai_hunter_add_package(ZLIB)
     if(DEPTHAI_ENABLE_BACKWARD)
-        hunter_add_package(Backward)
+        depthai_hunter_add_package(Backward)
     endif()
-    hunter_add_package(libnop)
+    depthai_hunter_add_package(libnop)
     if(DEPTHAI_PCL_SUPPORT)
-        hunter_add_package(jsoncpp)
+        depthai_hunter_add_package(jsoncpp)
     endif()
 endif()
 
